citrixitm/resource_citrixitm_dns_app.go

```diff
@@ -118,19 +123,21 @@ func update(id int, c *itm.Client, d *schema.ResourceData) error {
 			d.Get("fallback_cname").(string),
 			d.Get("app_data").(string),
 		)
-		log.Printf("[DEBUG] DNS app update options: %#v", opts)
-		_, err := c.DNSApps.Update(id, &opts, true)
+		log.Printf("[DEBUG] %s update options: %#v", resourceName, opts)
+		_, err := client.DNSApps.Update(id, &opts, true)
 		if err != nil {
-			return nil
+			log.Printf("[WARN] There was an error updating %s (id %d): %s", resourceName, id, err)
```

Comment:
shouldn't this result in an early return here instead of proceeding to `resourceCitrixITMDnsAppRead`?

Reply:
The canonical examples have the "update" function returning the result of the "read" function.

https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources

Reply:
cool